CMake/TestGNUCUDAVersion.cmake

```diff
@@ -0,0 +1,16 @@
+set(CUDA_GNU_COMPATIBLE TRUE)
+if(CMAKE_CXX_COMPILER_ID STREQUAL "GNU" AND CMAKE_CXX_COMPILER_VERSION VERSION_GREATER_EQUAL 12.0)
+  if(CMAKE_CUDA_COMPILER_VERSION VERSION_LESS_EQUAL 11.7)
+    set(CUDA_GNU_COMPLATIBLE FALSE)	  
+  endif()
+endif()
```

Comment:
If I use GCC 12 as the host compiler and set CMAKE_CUDA_HOST_COMPILER to GCC 11. This trap still stops me. So not good.

I think we just can't act on CMAKE_CXX_COMPILER at all since only CMAKE_CUDA_HOST_COMPILER matters.

Reply:
Does CMake provide a way to get information about the CMAKE_CUDA_HOST_COMPILER? I'd like to avoid parsing terminal output.

Reply:
If you mean get the version of host compiler beneath nvcc, I think the answer is no. Maybe we should just drop the version checking and only relies on the compilation check.